### PR TITLE
[PHPCS] Disallow line length > 400

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

PSR12 by default suggests to limit line lengths to 120.
I'm actually not sure if a limit of 120 is nowadays really needed. My screen is big enough to see a lot more 😀 
Anyway I think it could make sense to limit it to a certain amount of chars.

As a start this PR will limit it to 400 chars. That's a lot, but was a number that didn't cause too many required changes.
For now tests are excluded from the check, but we can discuss if it makes sense to include them as well.

@matomo-org/core-reviewers Which line length limit do you think make sense? I would suggest to aim a limit of 200
Once we have agreed on a certain limit we can start lowering the limit of the phpcs check step by step and adjust the required places...

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by michalkleiner** [comment]
My IDE with the side panels open and a font size comfortable to read on a 27'' screen displays 180 characters. If I make the panels slightly narrower it's ok to go to say 200 chars line length.

However, line length, in general, is not only about how many characters can fit on a screen, but it's also how much text can a person comfortably read on a line before they lose context and before multiline text becomes hard to follow (where the next line starts, skipping lines etc.).

It's an accessibility concern on its own and making the lines super long may make it harder to contribute. With that in mind, I would probably suggest 150 as a maximum.

If we were consistently getting over 150 due to e.g. control blocks nesting etc., it might be an indication to refactor rather than to increase the line length limit.

**Comment by caddoo** [comment]
Yeah the fewer characters per line the better for me, like @michalkleiner mentioned long lines are generally harder to read, and in addition probably a sign there is something you should refactor in your code if you are hitting 200 characters in each line.

When you are forced to write shorter lines, then your functions will then become taller which also can be a code smell that you should probably reduce the responsibility of the given function.

I generally have a really large font which helps me make sure I'm writing short easy to understand lines (and because I'm getting old).

150 characters seems reasonable.

**Comment by michalkleiner** [comment]
Updated the limit to 150, no additional changes seem to be needed, so once the tests pass we can merge this.

**Comment by michalkleiner** [comment]
And yet again I got caught out by not seeing the warnings locally, and of course there's more files to update.

**Comment by michalkleiner** [comment]
Back to 400 for now

**Comment by michalkleiner** [comment]
Test failures seem unrelated.

**Review by michalkleiner — APPROVED** [review]

**Review by mneudert — APPROVED** [review]
